service/rest_version.go

```diff
@@ -270,6 +272,19 @@ func (restapi restAPI) getVersionInfo(w http.ResponseWriter, r *http.Request) {
 		grip.Infof("adding BuildVariant %s", buildStatus.BuildVariant)
 	}
 
+	if evergreen.UseParserProject && destVersion.Config == "" {
```

Comment:
What happens when `evergreen.UseParserProject` is true but `getVersionInfo` is called on a version that is old enough not to have a parserProject?

This leads to a second question. I think there are cases in the UI where the yaml string is parsed. Do you know what path those will go through (if some versions have only yaml strings and others have only parserProjects on a single waterfall page)?

Reply:
To answer both these questions: in the case here that we're "using" parser project, we actually are just translating it to a config and putting it in the destVersion (we're also only doing this if the config isn't already populated, i.e. the first case you listed). So the UI is still only processing yaml strings.

Reply:
What about not returning a version at all? I don't really think it's useful for anyone. Whether you keep it as is or always set `destVersion.Config` to the empty string. lgtm.